httpcore/_backends/sync.py

```diff
@@ -54,12 +54,15 @@ def start_tls(
 
     def read(self, n: int, timeout: TimeoutDict) -> bytes:
         read_timeout = timeout.get("read")
-        exc_map = {socket.timeout: ReadTimeout, socket.error: ReadError}
+        exc_map = {socket.timeout: ReadTimeout, OSError: ReadError}
 
         with self.read_lock:
             with map_exceptions(exc_map):
                 self.sock.settimeout(read_timeout)
-                return self.sock.recv(n)
+                data = self.sock.recv(n)
+                if data == b"":
+                    raise OSError("Server disconnected while attempting read")
```

Comment:
Just a quick thought, will `self.sock` get closed elsewhere, e.g. in `__exit__` or when explicitly closing the client? I guess it's something you want to leave up to the client whether to close the socket in these cases.

Reply:
Yes, closing the socket is already handled by the connection pool, which closes all connections (which themselves close their sockets) on `__aexit__`.